src/dune_engine/scheduler.mli

```diff
@@ -59,15 +59,32 @@ module Run : sig
 
   exception Build_cancelled
 
-  (** Runs [once] in a loop, executing [finally] after every iteration, even if
-      Fiber.Never was encountered.
+  (** [poll once] runs [once] in a loop.
 
       If any source files change in the middle of iteration, it gets canceled.
 
       If [shutdown] is called, the current build will be canceled and new builds
       will not start. *)
   val poll : (unit, [ `Already_reported ]) Result.t Fiber.t -> unit Fiber.t
```

Comment:
I'd introduce a named type here so that we can easily see it the same between `poll` and `poll_passive`:
```suggestion
  type step = (unit, [ `Already_reported ]) Result.t Fiber.t
  val poll : step -> unit Fiber.t
```

Reply:
Sounds good!